### DISCUSSION TITLE
FOREIGN KEY constraint failed - Business shared folders

### DISCUSSION THREAD

**Discussion by laenrek** [abraunegg/onedrive#2014] [Q&A] [answered]
Hello,

I have successfuly compiled client version v2.4.19-1-gaf86b55, and synchronization of my own files and folders works fine.
However, the moment I try to synchronize remote shared folders, once I have added at leas one of my shared folders, I get this error:
```
$ onedrive --synchronize --sync-shared-folders --verbose --verbose
[...]
[DEBUG] The item we are syncing is a folder
[DEBUG] The reported parentId is not in the database. This potentially is a shared folder as 'item.driveId' != 'defaultDriveId'. Relevant Details: item.driveId (...), item.parentId (...)
[DEBUG] Update/Insert local database with item details

A database statement execution error occurred: FOREIGN KEY constraint failed

Please retry your command with --resync to fix any local database corruption issues.
```
Is there anything I am doing wrong? My system is Ubuntu 18.04.3 LTS
Thanks in advance

**Comment by abraunegg** [comment, marked as answer]
@laenrek 
> Is there anything I am doing wrong? My system is Ubuntu 18.04.3 LTS

There could be a number of issues here:
* You are using Ubuntu 18.x
  * Strongly advise you to upgrade to something more modern - Ubuntu 22.x ... do not use such an lethargic old version
* When `A database statement execution error occurred: FOREIGN KEY constraint failed` crops up, 99% of the time this is because, on older Ubuntu platforms, the 'onedrive' package has been installed. On Ubuntu 18.x you get a very horrid old version, that gets installed to a different path as compared to when you manually install the client ... so - these clients are conflicting with one another and the older one constantly will be writing over the database - not adding in the correct details - which is *why* explicitly this error occurs.
* You are using `--sync-shared-folders` flag which is for OneDrive Business Shared Folders - it is a *very* specific function - are you using a Business Account Type | SharePoint Account Type and have you configured this function as per this documentation: https://github.com/abraunegg/onedrive/blob/master/docs/BusinessSharedFolders.md
* If your Account Type is Personal, you need to add the Shared Folders to your account as per https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#shared-folders-onedrive-personal
* The last 1% is that you may have found a bug ... or there has been a recent API change by Microsoft ... 
  * Please follow the *correct* process to report a new bug with all the required respective details
  * Please read https://github.com/abraunegg/onedrive/#reporting-an-issue-or-bug very carefully if you are *worried* about the sensitivity of your debug log data


Questions for you to answer:
* What is your OneDrive Account Type?
* Can you upgrade your Ubuntu version to something a little more modern?
* Have you installed 'onedrive' by mistake - what does `sudo apt list --installed | grep 'onedrive'` detail?
  * If anything *has* been installed ... you must uninstall the package and reboot then reinstall the client from source
* Is your Account Type is Business or SharePoint and have you correctly configured OneDrive Shared Business Folders?
  * If not - ie - your account type is Personal Account Type, do not use the `--sync-shared-folders` flag it is pointless for you to use and add the Shared Folders to your account in the correct manner.

**Reply by laenrek** [reply]
I have installed a fresh arch linux (archlinux-2022.06.01-x86_64), and retried, but with the same result (plus a core dumped)
My account type is Business, and I get a list of folders following the document BusinessSharedFolders.md:
```
$ onedrive --list-shared-folders
Configuring Global Azure AD Endpoints
Initializing the Synchronization Engine ...

Listing available OneDrive Business Shared Folders:
---------------------------------------
Shared Folder:   Folder1
Shared By:       xx (xx@xx.com)
---------------------------------------
Shared Folder:   Folder2
Shared By:       xx (xx@xx.com)
```
Thanks.

**Reply by abraunegg** [reply]
@laenrek 
Please submit a bug report following the correct process
